{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "25fc4a04",
   "metadata": {},
   "source": [
    "# Combine all the quant.sf files into one gene expression matrix\n",
    "\n",
    "## 1. Creat a samples_metadata.csv file based on the existing salmon_output_* directories.\n",
    "\n",
    "cd /home/paperspace/BCHM5420_Project/Data/Data\n",
    "\n",
    "echo \"sample_id,condition,path\" > samples_metadata.csv\n",
    "\n",
    "for dir in salmon_output_*; do\n",
    "  sample=$(echo \"$dir\" | sed 's/salmon_output_//')\n",
    "\n",
    "  if [[ \"$sample\" == *FLT* ]]; then\n",
    "    condition=\"FLT\"\n",
    "  elif [[ \"$sample\" == *GC* ]]; then\n",
    "    condition=\"GC\"\n",
    "  else\n",
    "    condition=\"UNKNOWN\"\n",
    "  fi\n",
    "\n",
    "  echo \"$sample,$condition,$PWD/$dir/quant.sf\" >> samples_metadata.csv\n",
    "done\n",
    "\n",
    "## 2. Check the samples_metadata.csv file to make sure it's correct\n",
    "\n",
    "cat samples_metadata.csv\n",
    "\n",
    "## 3. Download and unzip the mouse gtf file, which will be needed in step 6\n",
    "\n",
    "cd /home/paperspace/BCHM5420_Project/Data/Data\n",
    "\n",
    "wget ftp://ftp.ensembl.org/pub/release-109/gtf/mus_musculus/Mus_musculus.GRCm39.109.gtf.gz\n",
    "gunzip Mus_musculus.GRCm39.109.gtf.gz\n",
    "\n",
    "## 4. Set up the R environment in VSCode\n",
    "\n",
    "R\n",
    "install.packages(\"BiocManager\")\n",
    "BiocManager::install(c(\"tximport\", \"readr\", \"DESeq2\", \"tximportData\"))\n",
    "\n",
    "library(tximport)\n",
    "library(readr)\n",
    "library(DESeq2)\n",
    "\n",
    "## 5. Read the sample metadata\n",
    "\n",
    "samples <- read.csv(\"samples_metadata.csv\")\n",
    "head(samples)\n",
    "\n",
    "## 6. Create a tx2gene mapping file to translate data from transcript to gene\n",
    "\n",
    "BiocManager::install(\"GenomicFeatures\")\n",
    "library(GenomicFeatures)\n",
    "\n",
    "txdb <- makeTxDbFromGFF(\"Mus_musculus.GRCm39.109.gtf\", format = \"gtf\")\n",
    "k <- keys(txdb, keytype = \"TXNAME\")\n",
    "tx2gene <- select(txdb, keys = k, columns = \"GENEID\", keytype = \"TXNAME\")\n",
    "\n",
    "write.csv(tx2gene, \"tx2gene.csv\", row.names = FALSE)\n",
    "\n",
    "## 7. Import counts using tximport\n",
    "\n",
    "files <- samples$path\n",
    "names(files) <- samples$sample_id\n",
    "\n",
    "txi <- tximport(files, type = \"salmon\", tx2gene = tx2gene, ignoreTxVersion = TRUE)\n",
    "\n",
    "# Now that the data has been summarized as gene-level counts, proceed to differential gene expression analysis using DESeq2\n"
   ]
  }
 ],
 "metadata": {
  "language_info": {
   "name": "python"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
